pyV2DL3/eventdisplay/util.py

```diff
@@ -70,9 +70,20 @@ def getGTI(BitArray, run_start_from_reference):
     """
 
     n = BitArray.size
-    TimeArray_s = [np.binary_repr(BitArray[i]).count("1") for i in range(n)]
-    duration_s = (n - 1) * 8 + TimeArray_s[-1]
-    ontime_s = np.sum(TimeArray_s[0:n])
+    TimeArray_b = ""
+    for i in range(n):
+
+        TimeArray_b = TimeArray_b + np.binary_repr(BitArray[i], width=8)[::-1]
+
+    nbits = len(TimeArray_b)
+    for i in range(nbits):
+        if (TimeArray_b[-1] == "0"):
```

Comment:
Not sure I understand this: here you are removing a trailing zero? Are there cases where there might be two trailing zeros?

Reply:
yes! I am removing trailing zeros at the end of array. It checks iteratively if the last element of array`TimeArray_b` is zero, and removes it from the array when zero is found, and loop breaks when the last element is 1.  

yes, there will be cases when there are more than one trailing zeros. For eg. see the case of `1162 20 0` for run number `74361` above in the conversation. This case have 22 trailing zeros.